internal/buildscripts/packaging/installer/install.ps1

```diff
@@ -689,28 +689,50 @@ restarted to apply the changes by restarting the system or running the following
     echo "$message"
 }
 
-if ($with_dotnet_instrumentation) {
-    echo "Installing SignalFx Dotnet Auto Instrumentation..."
-    Install-SignalFxDotnet
+$otel_resource_attributes = ""
+if ($deployment_env -ne "") {
+    echo "Setting deployment environment to $deployment_env"
+    $otel_resource_attributes = "deployment.environment=$deployment_env"
+} else {
+    echo "Deployment environment was not specified. Unless otherwise defined, will appear as 'unknown' in the UI."
+}
 
-    $regkey = "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager\Environment"
-    if ($deployment_env -ne "") {
-        echo "Setting SIGNALFX_ENV environment variable to $deployment_env ..."
-        update_registry -path "$regkey" -name "SIGNALFX_ENV" -value "$deployment_env"
+if ($with_dotnet_instrumentation) {
+    echo "Installing Splunk Distribution of OpenTelemetry .NET..."
+    $currentInstallVersion = Get-OpenTelemetryInstallVersion
+    if (-not $currentInstallVersion) {
+        Install-OpenTelemetryCore
     } else {
-        echo "SIGNALFX_ENV environment variable not set. Unless otherwise defined, will appear as 'unknown' in the UI."
+        Update-OpenTelemetryCore
```

Comment:
Here is a risk of doing unwanted silent update?

Reply:
Good question. Most of the time the instrumented applications are running in IIS and the update will stop and recycle them which seems what the typical user wants. However, if there are instrumented .NET applications outside IIS then the update will fail when copying files which probably is not very helpful. Let me think a bit more about this case.

Reply:
a283f827

Reply:
I'm thinking maybe to fail with a specific message "... dotnet is already installed, run the install script again with -UpdateDotnet to update the agent'" 🤔

Reply:
The warning message is `Splunk Distribution of OpenTelemetry .NET is already installed. Updating to the latest version...` and will be displayed before any error so I think it gives enough context in case of an error updating.